cookbook/controller/service.rst

```diff
@@ -232,6 +234,34 @@ inject *only* the exact service(s) that you need directly into the controller.
    are valid, exactly how you want to organize your reusable code is up to
    you.
 
+Base Controller Methods and their Service Replacements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This table explains how to replace the convenience methods of the base
+controller.
+
++-----------------------------+------------------------------------+----------------------------------------------------------------+
+| Method                      | Service                            |  PHP Code                                                      |
++=============================+====================================+================================================================+
+| ``createForm``              | ``form.factory``                   | ``$formFactory->create($type, $data, $options)``               |
+| ``createFormBuilder``       | ``form.factory``                   | ``$formFactory->createBuilder('form', $data, $options)``       |
+| ``createNotFoundException`` | -                                  | ``throw new NotFoundHttpException($message, $previous);``      |
+| ``forward``                 | ``http_kernel``                    | ``$httpKernel->forward($controller, $path, $query)``           |
+| ``generateUrl``             | ``router``                         | ``$router->generate($route, $params, $absolute)``              |
+| ``getDoctrine``             | ``doctrine``                       |                                                                |
+| ``isGranted``               | ``security.authorization_checker`` | ``$authorizationChecker->isGranted($attributes, $object);``    |
```

Comment:
this should be ``security.context`` here (this is for 2.3). It should only be changed to ``security.authorization_checker`` when merging things into 2.6